lib/cretonne/meta/isa/intel/encodings.py

```diff
@@ -142,38 +159,52 @@ def enc_flt(inst, recipe, *args, **kwargs):
 I64.enc(base.ctz.i32, *r.urm.rex(0xf3, 0x0f, 0xbc), isap=cfg.use_bmi1)
 I64.enc(base.ctz.i32, *r.urm(0xf3, 0x0f, 0xbc), isap=cfg.use_bmi1)
 
+#
 # Loads and stores.
-I32.enc(base.store.i32.any, *r.st(0x89))
-I32.enc(base.store.i32.any, *r.stDisp8(0x89))
-I32.enc(base.store.i32.any, *r.stDisp32(0x89))
+#
+enc_i32_i64_ld_st(base.store, 1, r.st, 0x89)
+enc_i32_i64_ld_st(base.store, 1, r.stDisp8, 0x89)
+enc_i32_i64_ld_st(base.store, 1, r.stDisp32, 0x89)
+
+I64.enc(base.istore32.i64.any, *r.st.rex(0x89))
+I64.enc(base.istore32.i64.any, *r.stDisp8.rex(0x89))
+I64.enc(base.istore32.i64.any, *r.stDisp32.rex(0x89))
+
+enc_i32_i64_ld_st(base.istore16, 0, r.st, 0x66, 0x89)
+enc_i32_i64_ld_st(base.istore16, 0, r.stDisp8, 0x66, 0x89)
+enc_i32_i64_ld_st(base.istore16, 0, r.stDisp32, 0x66, 0x89)
+
+enc_i32_i64_ld_st(base.istore8, 0, r.st_abcd, 0x88)
+enc_i32_i64_ld_st(base.istore8, 0, r.stDisp8_abcd, 0x88)
+enc_i32_i64_ld_st(base.istore8, 0, r.stDisp32_abcd, 0x88)
```

Comment:
Ok, got it. I won't use `enc_i32_i64_ld_st` for `istore8` but rather declare the `I32.enc` and `I64.enc` directly with the correct recipes.